src/services/open-api-client.js

```diff
@@ -34,26 +35,21 @@ class OpenApiClient {
     const params = this.getParams(opts, operation);
 
     if (!opts.uri) {
-      opts.uri = this.getUri(opts);
+      opts.uri = this.getUri(opts.uri, opts);
     }
 
     // If the URI is relative, determine the host and prepend it.
     if (opts.uri.startsWith('/')) {
       if (!opts.host) {
         opts.host = path.server;
       }
-
-      if (opts.region) {
-        const parts = opts.host.split('.');
-
-        // From 'https://api.twilio.com/' to 'https://api.{region}.twilio.com/'
-        if (parts.length > 1 && parts[1] !== opts.region) {
-          parts.splice(1, 0, opts.region);
-          opts.host = parts.join('.');
-        }
-      }
-
+      opts.host = this.getHost(opts.host, opts);
       opts.uri = opts.host + opts.uri;
+    } else if (opts.uri) {
+      let uri = new url.URL(opts.uri);
+      uri.hostname = this.getHost(uri.hostname, opts);
+      uri.pathname = this.getUri(uri.pathname, opts);
+      opts.uri = uri.href;
```

Comment:
I think this logic can all be pulled out of the block. If we've made it here, `opts.uri` should be fully formed and we can now use the generic logic of parsing it and updating/inserting the edge and region.

Reply:
Nice catch